packages/gestalt/src/Button.js

```diff
@@ -76,8 +77,17 @@ export default function Button(props: Props) {
       </Text>
     </button>
   );
+  /* eslint-enable react/button-has-type */
 }
 
+Button.defaultProps = {
```

Comment:
I'm so torn on this one. None of our tools really introspect `defaultProps` and it's not a required part of the React API so this definitely feels like extra typing...

Reply:
Agreed - I started off by playing with the typing and defaulting, hoping to fix the `type={type}` error but to no avail since I found the github thread linked above saying it needs to be hardcoded to `"button"` or `"submit"`. Figured I would keep the `defaultProps` that I created anyways since it is a common pattern. Thoughts, I can revert the defaulting since the rule needs to be disabled anyways